sklearn/preprocessing/_polynomial.py

```diff
@@ -973,21 +1019,31 @@ def transform(self, X):
                 # Only the first degree and last degree number of splines
                 # have non-zero values at the boundaries.
 
-                # spline values at boundaries
-                f_min = spl(xmin)
-                f_max = spl(xmax)
                 mask = X[:, i] < xmin
                 if np.any(mask):
-                    XBS[mask, (i * n_splines) : (i * n_splines + degree)] = f_min[
-                        :degree
-                    ]
+                    if use_sparse:
+                        # Note: See comment about SparseEfficiencyWarning above.
+                        if not sparse.isspmatrix_lil(XBS_sparse):
+                            XBS_sparse = XBS_sparse.tolil()
```

Comment:
Looking at [the implementation of `scipy.sparse.lil_matrix.tolil`](https://github.com/scipy/scipy/blob/a9ee25e0f47d5ff1d0ae8c36e165cacd1f966b7e/scipy/sparse/_lil.py#L439-L443), I think this can be boiled down to:
```suggestion
                        XBS_sparse = XBS_sparse.tolil()
```